porting: Update the custom targets guide for CMake and Mbed CLI 2

Patater opened the pull request:
The custom targets documentation didn't cover how to make your custom
target buildable with CMake. Update it to cover all aspects of adding a
custom target such that it can be build using Mbed OS's CMake build
system.

Patater committed:
porting: Update the custom targets guide for CMake

The custom targets documentation didn't cover how to make your custom
target buildable with Mbed CLI 2 or CMake. Update it to cover all
aspects of adding a custom target such that it can be build using Mbed
OS's CMake build system.

urutva commented on docs/porting/custom-target-howto.md:
```diff
@@ -0,0 +1,117 @@
+├── CMakeLists.txt
+├── CONTRIBUTING.md
+├── custom_targets.json
+├── IMAGINARY              <--- The new folder with new files in it!
```
Currently, we follow `TARGET_<BOARD_NAME>` naming convention for board/target related files. Can we follow the same convention in the example as well?

Patater replied:
The targets aren't named TARGET_BLAH. The target name is actually just BLAH. For folder names, given the limitation of the old tools, we did name things TARGET_BLAH. I'll do that here for old time's sake.

urutva commented on docs/porting/custom-target-howto.md:
```diff
@@ -0,0 +1,117 @@
+    ├── CMakeLists.txt
+    ├── PeripheralNames.h
+    ├── PeripheralPins.h
+    └── PinNames.h
```
From [the Mbed OS porting guide](https://os.mbed.com/docs/mbed-os/latest/porting/porting-custom-boards.html), `PeripheralPins.c` should also be added here.

Patater replied:
Many of these are optional.

urutva commented on docs/porting/custom-target-howto.md:
````diff
@@ -0,0 +1,117 @@
+    └── official_armmbed_example_badge.png
+```
+
+The CMakeLists.txt file should have at a minimum a new library target, as well as a list of which files or include directories your target needs. The [`target_link_libraries()`](https://cmake.org/cmake/help/latest/command/target_link_libraries.html) directive should specify which MCU CMake target your custom board depends on, so you'll enter here the MCU Cmake target name you noted previously.
````
```suggestion
The ./IMAGINARY/CMakeLists.txt file should have at a minimum a new library target, as well as a list of which files or include directories your target needs. The [`target_link_libraries()`](https://cmake.org/cmake/help/latest/command/target_link_libraries.html) directive should specify which MCU CMake target your custom board depends on, so you'll enter here the MCU Cmake target name you noted previously.
```

urutva replied:
Would adding the folder name helpful?

Patater replied:
Good idea

Patater replied:
I've removed this file, so this probably isn't relevant anymore.

Patater committed:
porting: Wrap lines and remove trailing whitespace

Hard-wrap all too long lines and remove all trailing whitespace.

Patater committed:
porting: Update custom targets guide for CLI 2

Update our custom targets porting guide to use Mbed CLI 2 instead of
Mbed CLI 1.

Patater committed:
porting: Inherit from MCU_STM32L475xG

The porting guide was written with a previous target factoring in Mbed
OS. The targets have been refactored since then, so the old instructions
don't work. Update the json and instructions so the guide has a
possibility of working.

Patater committed:
porting: Fix formatting of json names

Patater committed:
porting: Use / instead of \ consistently

Patater committed:
porting: Update to Thread::sleep_for()

The wait_ms() API has been removed from Mbed OS. Update the porting
guide to use Thread::sleep_for() instead.

Patater committed:
porting: Sort the custom target file list

The list of files to expect isn't in the same order that an `ls` on the
command line would be in. Sort the list.
